packages/pyright-internal/src/languageService/signatureHelpProvider.ts

```diff
@@ -141,13 +145,32 @@ export class SignatureHelpProvider {
                 paramName = params[params.length - 1].name || '';
             }
 
+            // If we have a typedKwargs, the param name will be wrong.
+            const kwargsIndex = paramIndex >= params.length ? params.length - 1 : paramIndex;
+            const kwargsParam = params[kwargsIndex];
+            if (
+                isTypedKwargs(kwargsParam) &&
+                isClassInstance(kwargsParam.type) &&
```

Comment:
Similarly, I think this `isClassInstance` check is unnecessary?

Reply:
Same reason for this one. Although maybe I can add a 'type is Class' to isTypedKwargs

Reply:
No that doesn't work because it's not passing the type into that function. So yeah this is necessary to get typescript to understand that the `.type.details` member actually exists.